hibernate-core/src/main/java/org/hibernate/hql/spi/id/inline/AbstractInlineIdsDeleteHandlerImpl.java

```diff
@@ -42,6 +43,9 @@ public AbstractInlineIdsDeleteHandlerImpl(
 
 	@Override
 	public String[] getSqlStatements() {
+		if ( deletes.isEmpty() ) {
+			return new String[0];
```

Comment:
Do we allow the returned value to be modifiable? If not, returning `Collections.emptyList()` would be more efficient and less confusing. The same suggestion applies for the later empty collection returned. Minor compared to the issue solved in this PR.

Reply:
we method return type is an array

Reply:
my fault. we might use `ArrayHelper.EMPTY_STRING_ARRAY`, though not a big deal.

Reply:
done and thanks a lot for the review :)

Reply:
If we're nitpickin for optimisations, I might add that it's even faster to remove the `if` check  ;)

Reply:
I'm joking, but let me clarify why: it's because I would not expect the `deleted` list to ever be empty, so the check is redundant.